.github/workflows/e2e_aks_runtime.yml

```diff
@@ -0,0 +1,55 @@
+name: e2e test aks runtime
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "16 6 * * 6" # 6:16 on Saturdays
+  pull_request:
+    paths:
+      - e2e/aks-runtime/**
+
+jobs:
+  # steps taken from https://learn.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=apt#option-2-step-by-step-installation-instructions
+  install-software:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Install `az` with extensions
+        run: |
+          sudo apt-get update
+          sudo apt-get install apt-transport-https ca-certificates curl gnupg lsb-release
+          sudo mkdir -p /etc/apt/keyrings
+          curl -sLS https://packages.microsoft.com/keys/microsoft.asc |
+          gpg --dearmor | sudo tee /etc/apt/keyrings/microsoft.gpg > /dev/null
+          sudo chmod go+r /etc/apt/keyrings/microsoft.gpg
+          AZ_DIST=$(lsb_release -cs)
+
+          sudo tee /etc/apt/sources.list.d/azure-cli.sources <<EOF
+          Types: deb
+          URIs: https://packages.microsoft.com/repos/azure-cli/
+          Suites: ${AZ_DIST}
+          Components: main
+          Architectures: $(dpkg --print-architecture)
+          Signed-by: /etc/apt/keyrings/microsoft.gpg
+          EOF
+
+          sudo apt-get update
+          sudo apt-get install azure-cli
+
+          az extension add --name aks-preview
+          az extension add --name confcom
+
+  test_matrix:
+    name: Test aks runtime
+    needs: install-software
```

Comment:
If this is meant to run the e2e test with the az cli installed in `install-software`, this doesn't work. `install-software` runs in another job, in another VM, and any changes there don't carry over to this job. If we want the changes to persist, all steps will have to be part of the same job.